multiapps-controller-process/src/test/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceCreateUpdateServiceActionsStepTest.java

```diff
@@ -71,6 +75,31 @@ void testExecute(String inputFilename, String expectedExceptionMessage) {
         validateActions(input);
     }
 
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void testServiceParametersFetchingErrorHandling(boolean isOptionalService) {
+        var service = createMockServiceInstance(isOptionalService);
+        context.setVariable(Variables.SERVICE_TO_PROCESS, service);
+        context.setVariable(Variables.SERVICE_KEYS_TO_CREATE, Map.of());
+
+        Mockito.when(client.getServiceInstanceParameters(Mockito.any(UUID.class)))
+               .thenThrow(new CloudOperationException(HttpStatus.INTERNAL_SERVER_ERROR));
+        Mockito.when(client.getServiceInstance("service", false))
+               .thenReturn(service);
+
+        if (isOptionalService) {
+            step.execute(execution);
+            assertStepIsRunning();
+            List<ServiceAction> serviceActionsToExecute = context.getVariable(Variables.SERVICE_ACTIONS_TO_EXCECUTE);
+            assertTrue(serviceActionsToExecute.contains(ServiceAction.UPDATE_CREDENTIALS),
+                       "Actions should contain " + ServiceAction.UPDATE_CREDENTIALS);
+            return;
+        }
+        SLException exception = assertThrows(SLException.class, () -> step.execute(execution));
+        assertTrue(exception.getCause() instanceof CloudOperationException);
+        assertSame(((CloudOperationException) exception.getCause()).getStatusCode(), HttpStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Why don't use `assertEquals`?

Reply:
The status code is an enum variable, not an object. The `==` operator is sufficient for checking equality. This method was a suggestion by the IDE.